### PR TITLE
Fix ndmin issue 10063

### LINKED ISSUE #10063: Quantity ndmin argument is ignored if constructed from another quantity with convertible unit
### Description
If a `Quantity` is constructed from another quantity that has a different but convertible unit, the `ndmin` argument is ignored.

### Expected behavior
`ndmin` argument is not ignored

### Steps to Reproduce
```python
>>> import astropy.units as u
>>> u.Quantity(u.Quantity(1, 'm'), 'm', ndmin=1).ndim
1
>>> u.Quantity(u.Quantity(1, 'cm'), 'm', ndmin=1).ndim
0
```

### PR DESCRIPTION
**Pull request by MaxNoe**
### Description

Fixes #10063

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -99,7 +99,6 @@ asdf_schema_tests_enabled = true
 xfail_strict = true
 qt_no_exception_capture = 1
 filterwarnings =
-    error
```
Comment (bsipocz) [inline setup.cfg]: please don't remove it

Comment (pllim) [inline setup.cfg]: This change is unrelated. Please revert.

Comment (MaxNoe) [reply setup.cfg]: oh sorry, accidentally commited

Comment (MaxNoe) [reply setup.cfg]: Btw. this was the warning I got:
```
pytest.PytestDeprecationWarning: direct construction of DocTestModulePlus has been deprecated, please use DocTestModulePlus.from_parent
```

Comment (bsipocz) [reply setup.cfg]: do you see this locally? If yes, then downgrade pytest version to <5.4

Comment (MaxNoe) [reply setup.cfg]: yes

File: CHANGES.rst [hunk 2]
```diff
@@ -103,6 +103,9 @@ astropy.units
 - Added integrated flux unit conversion to ``spectral_density`` equivalency.
   [#10015]
 
+- Fixed ``u.Quantity`` not taking into account ``ndmin`` if constructed from
+  another ``u.Quantity`` instance with different but convertible unit [#10066]
```
Comment (pllim) [inline CHANGES.rst]: Thanks! Change log should be under a bug fix release but I'll let @bsipocz decide whether it should be 4.0.1 or 4.0.2.

I'll also let others to decide if this needs squashing. 😬

Comment (bsipocz) [reply CHANGES.rst]: either would work, I can move around stuff this close to the release

Comment (MaxNoe) [reply CHANGES.rst]: I moved it to 4.0.1

**Comment by MaxNoe** [comment]
@mhvk ok, done

**Comment by MaxNoe** [comment]
@mhvk tue codecov seems inconclusive.  Should I do something or can we merge as is?

**Comment by pllim** [comment]
You can probably ignore the codecov failure. 🤷‍♀

**Comment by bsipocz** [comment]
Thanks @MaxNoe!

**Comment by mhvk** [comment]
yes, that would seem a false positive. Merging! ...  Oops, @bsipocz was ahead of me!
